### ISSUE TITLE
Indexes are being created twice

### ISSUE THREAD

**Issue by Sahara150** [techouse/mysql-to-sqlite3#20]
**Describe the bug**
I have a MySQL database with some indexes. When I convert it to SQLite every index is being created twice. Once with "indexname" and once with "tablename_indexname".
This results in an invalid scheme for Android Room, since there are too many indexes, which makes it quite problematic.

**Expected behaviour**
Index should only be created once or there should be an option, that disables the doubled creation. (In case some programs need the "tablename_indexname" naming.)

**System Information**

```bash
$ mysql2sqlite --version
```

```
| software               | version                              |
|------------------------|--------------------------------------|
| mysql-to-sqlite3       | 1.4.4                                |
|                        |                                      |
| Operating System       | Windows 10                           |
| Python                 | CPython 3.8.2                        |
| MySQL                  | MySQL client not found on the system |
| SQLite                 | 3.28.0                               |
|                        |                                      |
| click                  | 8.0.1                                |
| mysql-connector-python | 8.0.26                               |
| python-slugify         | 5.0.2                                |
| pytimeparse            | 1.1.8                                |
| simplejson             | 3.17.3                               |
| six                    | 1.16.0                               |
| tabulate               | 0.8.9                                |
| tqdm                   | 4.62.0                               |```


  + MariaDB 10.4.11 (with XAMPP)

**Comment by techouse** [comment]
Can you supply me with the DDL of your MySQL database?

**Comment by Sahara150** [comment]
Can´t find a ddl. Data folder containts .frm and .ibd files.
The indices I got from the query were:
<html>
<body>


index_Articles_germanArticle | 0 | 0 | germanArticle
-- | -- | -- | --
PRIMARY | 1 | 0 | id


</body>
</html>
So no 2nd index returned from the query.
Looking at my current results from Room I have the suspicion, that they are creating missing indices themselves. (After deleting all indices, the right ones are there.) So the issue probably is not that the index is created twice, but only, that the table-name is appended to the index in your code and that´s why Room thinks the index is missing/not there (can´t find the name) and creates a new one.
But it would be good, when there is an option to disable the appending of the table name.
I found the line, where you are doing it:
https://github.com/techouse/mysql-to-sqlite3/blob/8d7f96f71a34a6cd80bf14e27d2ad98230808401/mysql_to_sqlite3/transporter.py#L320
You are defining the name of the index as "tablename_indexname".
The indices can´t be deleted completely, because they are needed to define the foreign keys again.

**Comment by techouse** [comment]
Hey,

A DDL is short for "data definition language", basically your `CREATE TABLE ...` statement.

Prefixing all the indices with table names ensures unique names, otherwise we can get collisions, so to remove that would make it unstable.

You should open the SQLite up in a DB editor am do it manually.

**Comment by Sahara150** [comment]
Aaah, that´s what happens, when you are thinking of "dll"s when there is written "DDL". :D 

I get that, I am not asking you to remove it. I am asking for an option in the CLI to disable that. So there is the possibility to preserve the old index names with a hint in the options parameter, that this shall only be used with guaranteed unique indices. 

This would give more flexibility, since Room e.g. appends the table name after "index" like "index_tableName_columnAffected" and not before it. 

For people that need to fulfill a predefined scheme this flexibility is necessary.

I tried to manually adapt the indices, but sadly my compiler found out about it and was complaining that the Database is manipulated. :D

**Comment by techouse** [comment]
Sure, I can add that CLI option.

**Comment by techouse** [comment]
Added `-K`, `--prefix-indices` CLI option to prefix indices with table names.
This used to be the default behavior until now. To keep the old behavior simply use this CLI option.

Released in [v1.4.5](https://github.com/techouse/mysql-to-sqlite3/releases/tag/v1.4.5)
